ivy/functional/ivy/device.py

```diff
@@ -163,6 +154,20 @@ def _get_nvml_gpu_handle(device: Union[ivy.Device, ivy.NativeDevice], /) -> int:
     return handle
 
 
+def shift_native_arrays_on_def_dev(*args, dst_dev=None, **kwargs):
+    with ivy.ArrayMode(False):
+        default_device = ivy.default_device(dst_dev, as_native=True)
+        args, kwargs = ivy.nested_map(
+            [args, kwargs],
+            lambda x: (
+                ivy.to_device(x, default_device)
+                if (ivy.is_native_array(x) and ivy.dev(x) != default_device)
```

Comment:
Also, not completely sure why we need to explicitly check for native arrays, are ivy arrays being handled in the decorator itself? If that's the case we could probably also check for native arrays in the decorator

Reply:
I'm not sure what you mean. we're checking for native arrays here because we're passing `args` and `kwargs` to `nested_map` so there can be other input arguments in there.

Reply:
Oh ignore that, I probably misunderstood what this line was doing, given that it's directly used in a backend implementation it doesn't need to check for `ivy` arrays using `is_ivy_array` or anything like that. Thanks for clarifying!